docs/web/sdk-component-reference.mdx

```diff
@@ -0,0 +1,384 @@
+
+    protected _getSerializableProperties(): PropertyDefs<
+        ExampleComponentModelProperties
+    > {
+        // highlight-next-line
+        const props = super._getSerializableProperties();
+        return {
+            ...props,
+            items: {
+                serializeModes: ["initial"],
+                default: ["Default Value"],
+                serialize: () => this.items.toArray(),
+                deserialize: (items) => {
+                    this.items.removeAll();
+                    this.items.addMany(items);
+                },
+            },
+        };
+    }
+}
+```
+
+### Component Defaults
+
+Most components support the [`config` attribute](concepts-app-config.mdx#linking-layout-to-app-config) in the layout, which links a component model to configuration in the app config JSON. However, many component models have default values they can supply for initialization instead of relying on configuration. This means that if you omit the `config` property for certain components, the component model will attempt to create itself with its default values. An example of defining these default values can be seen in [this example](sdk-serializable-properties.mdx#populating-the-class-with-configuration-values). It's also the mechanism that powers the default map displayed by this layout.
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<layout xmlns="https://geocortex.com/layout/v1">
+    <map/> <!-- The map component is populated with values from the MapModel's `_getSerializableProperties` function. -->
+</layout>
+```
+
+### Initialization and Teardown
+
+Component models have an initialization method, which can be used to perform asynchronous startup logic, and a teardown method, which can be used to free resources when a component is removed from the layout.
+
+:::important
+Always call `super._onInitialize()` when overriding the `_onInitialize` method.
+:::
+
+```ts
+export default class ExampleComponentModel extends ComponentModelBase {
+    items: Collection<string> = new Collection<string>();
+
+    protected async _onInitialize(): Promise<void> {
+        // highlight-next-line
+        super._onInitialize();
+
+        this.items.add("1");
+        this.items.add("2");
+        this.items.add("3");
+        this.items.add("4");
+        // ...other initialization logic
+    }
+
+    destroy(): void {
+        this.items.destroy();
+        // ...free up resources
+    }
+}
+```
+
+## Component Lifecycle
+
+When a Geocortex Web Application boots up, the set of components which are initially active in the layout are created and initialized. [Components like the `Panel`](api-layout-core-components.mdx#panels) will only activate their first child. All initially inactive components will only be created when they are activated in the layout.
+
+:::tip
+The `UIService` contains methods like `getActiveChildComponentIds()` which can be used to examine and manage the active state of components and their children.
```

Comment:
Is this something we want to promote? @eric-parsons would be best to answer I think.

Reply:
Container components very often need to react to activation of their children. For example panel, tabs, taskbar, expand, IWTM, etc. might need to change the active view or open a popout. While I don't love the design of the current API, right now it's the only way to handle these use cases.

When I first read the sentence "can be used to examine and manage the active state of components and their children", it sounded like you can directly get a hold of the child models and manipulate their state, which would be an anti-pattern. I think "activation state" might be better?

Reply:
Rephrased to better reflect how activation state is manipulated through the UI Service.